### DISCUSSION TITLE
What's the release plan with the new main-dev split?

### DISCUSSION THREAD

**Discussion by TeutonJon78** [vladmandic/sdnext#1601] [General]
It's great for release management to have the split between dev and main so that breakages happen less often for people when updating, so thanks! There seems to be a lot of exciting stuff going on in the dev branch right now!

However, I'm curious what is the strategy for when to push dev to main? Time based? Certain criteria based?

And on another side note, is there any place we can track the progress on integration effort with Anapnoe's UI (if still in the works). Not asking for an ETA, since we'll get it when we get it.

**Comment by vladmandic** [comment]
separation of `dev` vs `main` is always good, but its question of resources - i just don't have cycles to maintain both all the time.
so for now, intention is to:
1. bugifxes and minor features go directly into `main`
2. large features that require multiple commits and explicit testing will go into a feature branch, for example `diffusers`
  feature branch merge with `main` results goes in `dev` so it can stay there until everything is cleaned up and tested  
  as such, dev will not be active branch all the time, it will be used as-needed.
3. `dev` is merged into `main`

**Comment by TeutonJon78** [comment]
Thanks for the answer. It's definitely a balance act between better release management and time spent, even more so for smaller projects. 

Thanks for the work you and the other main devs do!
